File: src/main/java/kr/co/bookand/backend/bookstore/repository/BookStoreRepository.java
package kr.co.bookand.backend.bookstore.repository;

import kr.co.bookand.backend.bookstore.domain.BookStore;
import kr.co.bookand.backend.common.domain.Status;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface BookStoreRepository extends JpaRepository<BookStore, Long>, BookStoreRepositoryCustom {

    Page<BookStore> findAllByStatus(Status status, Pageable pageable);

    List<BookStore> findAllByStatus(Status status);

    boolean existsByName(String name);

    Long countAllByVisibility(boolean visibility);

    Long countAllByVisibilityAndStatus(boolean visibility, Status status);
}

File: src/main/kotlin/kr/co/bookand/backend/bookstore/repository/BookstoreRepository.kt
package kr.co.bookand.backend.bookstore.repository

import kr.co.bookand.backend.bookstore.domain.Bookstore
import kr.co.bookand.backend.common.Status
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository

interface BookstoreRepository: JpaRepository<Bookstore, Long>, BookstoreRepositoryCustom {

    fun existsByName(name: String): Boolean
    fun countAllByVisibility(visibility: Boolean): Long
    fun countAllByVisibilityAndStatus(visibility: Boolean, status: Status): Long
    fun findAllByStatus(status: Status?, pageable: Pageable): Page<Bookstore>

    fun findAllByStatus(status: Status): List<Bookstore>


}